source/state_representation/test/tests/test_cartesian_state.cpp

```diff
@@ -460,6 +461,17 @@ TEST(CartesianStateTest, TestImplicitConversion) {
   tf1 += vel;
 }
 
+TEST(CartesianStateTest, TestStateClamping) {
+  CartesianState state = CartesianState::Identity("test");
+  EXPECT_THROW(state.clamp_state_variable(1, CartesianStateVariable::ORIENTATION), exceptions::NotImplementedException);
+  EXPECT_THROW(state.clamp_state_variable(1, CartesianStateVariable::POSE), exceptions::NotImplementedException);
+  state.set_linear_velocity(Eigen::Vector3d(-2.0, 1, 5));
+  state.clamp_state_variable(3.0, CartesianStateVariable::LINEAR_VELOCITY);
+  EXPECT_EQ(state.get_linear_velocity().norm(), 3.0);
+  state.clamp_state_variable(10.0, CartesianStateVariable::LINEAR_VELOCITY, 0.5);
+  EXPECT_EQ(state.get_linear_velocity().norm(), 0.0);
```

Comment:
You have a test for the upper clamping as well as the lower noise threshold. I would suggest an additional test that makes sure `clamp` has no effect when the norm is above the noise ratio and below the max threshold.